packages/edit-site/src/components/header/document-actions/index.js

```diff
@@ -20,6 +20,9 @@ function getBlockDisplayText( block ) {
 		: null;
 }
 
+// We want to show information in the header if we are editing a "template area."
+const TEMPLATE_AREAS = [ 'core/template-part', 'core/post-content' ];
```

Comment:
`Non-blocking`:

There were discussions to rename template part to template area https://github.com/WordPress/gutenberg/issues/24853. To avoid potential conflicts, would something like `TEMPLATE_LIKE_BLOCKS` work here as well?